javav2/example_code/transcribe-streaming/src/main/java/com/amazonaws/transcribestreaming/TranscribeStreamingDemoFile.java

```diff
@@ -1,199 +1,199 @@
-
-                if (len <= 0) {
-                    audioBuffer = ByteBuffer.allocate(0);
-                } else {
-                    audioBuffer = ByteBuffer.wrap(audioBytes, 0, len);
-                }
-            } catch (IOException e) {
-                throw new UncheckedIOException(e);
-            }
-
-            return audioBuffer;
-        }
-
-        private AudioEvent audioEventFromBuffer(ByteBuffer bb) {
-            return AudioEvent.builder()
-                    .audioChunk(SdkBytes.fromByteBuffer(bb))
-                    .build();
-        }
-    }
-}
-// snippet-end:[transcribe.java-streaming-demo-file.main]
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.transcribestreaming;
+
+// snippet-start:[transcribe.java-streaming-demo-file.import]
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.transcribestreaming.TranscribeStreamingAsyncClient;
+import software.amazon.awssdk.services.transcribestreaming.model.*;
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicLong;
+// snippet-end:[transcribe.java-streaming-demo-file.import]
+
+// snippet-start:[transcribe.java-streaming-demo-file.main]
+/**
+ * To run this AWS code example, ensure that you have set up your development
+ * environment, including your AWS credentials.
+ *
+ * For information, see this documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ */
+
+public class TranscribeStreamingDemoFile {
+    private static final Region REGION = Region.US_EAST_1;
+    private static TranscribeStreamingAsyncClient client;
+
+    public static void main(String args[]) throws ExecutionException, InterruptedException {
+
+        final String USAGE = "\n" +
+                "Usage:\n" +
+                "    <file> \n\n" +
+                "Where:\n" +
+                "    file - the location of a PCM file to transcribe. In this example, ensure the PCM file is 16 hertz (Hz). \n";
+
+        if (args.length != 1) {
+            System.out.println(USAGE);
+            System.exit(1);
+        }
+
+        String file = args[0];
+        client = TranscribeStreamingAsyncClient.builder()
+                .region(REGION)
+                .build();
+
+        CompletableFuture<Void> result = client.startStreamTranscription(getRequest(16_000),
+                new AudioStreamPublisher(getStreamFromFile(file)),
+                getResponseHandler());
+
+        result.get();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you are using `CompletableFuture.get()` or `Future.get()` without setting its timeout parameter. Be sure to set timeout values or handle timeout exceptions to prevent threads from waiting for a process to end. This can lead to downtime for your customers. For more information, see [CompletableFuture](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get-long-java.util.concurrent.TimeUnit-) and [Future](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html#get-long-java.util.concurrent.TimeUnit-) documentation.

Suggested remediation:
Set timeout parameters for `CompletableFuture.get()` or `Future.get()`.Exceptions, such as `ExecutionException`, `InterruptedException` and `TimeoutException` are caught and errors are logged.
```
@@ -59,3 +59,14 @@
 
-        result.get();
+        try {
+            result.get(100, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException ie) {
+            log.error(&#34;Thread interrupted.&#34;, ie);
+            throw new InterruptedException(ie.getMessage());
+        } catch (ExecutionException xe) {
+            log.error(&#34;The task aborted when attempting to retrieve its result.&#34;, xe);
+            throw new ExecutionException(xe.getMessage());
+        } catch (TimeoutException te) {
+            log.error(&#34;Future is not done when timeout.&#34;, te);
+            throw new TimeoutException(te.getMessage());
+        }
         client.close();

```